lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -137,57 +162,159 @@ static LogicalResult createPoolingOp(
                              ValueRange{paddedInput, windowTensor},
                              outTensorInitialized, stridesAttr, dilationAttr)
                .getResult(0);
-
   return success();
 }
 
-
 namespace {
-class ConvertAtenMaxPool2dOp : public OpConversionPattern<AtenMaxPool2dOp> {
+template <typename OpTy, int Dim>
```

Comment:
I don't think we need the `Dim` parameter. You can use `if (std::is_same<OpTy, AtenMaxPool2dOp>()) ...`. This also makes it more explicit in the code how each op is being handled.

Reply:
Good idea! Fixed!